### DISCUSSION TITLE
simulation of 2D cases

### DISCUSSION THREAD

**Discussion by ZYuan123** [peridigm/peridigm#299] [Q&A] [answered]
Hi

Is Peridigm working in 2D simulations with only one layer of particles?  since only 3D example are shown in its' codes.

If it works, how can I defined 2D case in input desk, is that defined by "plane stress true"? the critical stretch will be given by 2D equation or 3D equation?

hope anyone give some help.

regards,
Yuan

**Comment by ZYuan123** [comment]
I have this question because this is a user's guide[1] said "Currently, Peridigm only uses the 3D formulation of the peridynamics, e.g. for surface correction factors. Therefore, only the 3D equations shall be used for material mod-elling."


[1] Martin Rädel, Christian Willberg, Peridigm Users Guide, DLR-IB-FA-BS-2018-23, DLR Report, 2018, in PeriDoX, DOI: 10.5281/zenodo.1403015

**Comment by djlittl** [comment, marked as answer]
Peridigm doesn't directly support plane stress or plane strain.  There is nothing preventing you from having a discretization with a single layer of particles, but you would need to alter the formulation in the material model to get a reasonable result.  This would not be difficult to do, generally speaking.  There are papers in the literature that show how to modify certain material models in accordance with the plane strain or plane stress assumption.  I think you basically just need to change the constants.

**Reply by jritta** [reply]
As @djlittl already mentioned it mainly depends on how the material constants are defined. For the `Elastic Bond-Based` material the plane stress/strain is already implemented (see the two 2D tests for the [`Elastic Bond-Based` material model](https://github.com/peridigm/peridigm/tree/master/test/verification/ElasticBondBased)). For the `Elasitc` (state based) material model you can implement it in a similar fashion. The details you can find for example in the book [Handbook of Peridynamic Modeling](https://doi.org/10.1201/9781315373331) pp. 152.

Best,
Jonas

**Reply by ZYuan123** [reply]
@djlittl @jritta  
Sincerely thank for your reply. that is exactly what i need. helped a lot.
